frame/support/src/weights.rs

```diff
@@ -530,6 +533,90 @@ impl RuntimeDbWeight {
 	}
 }
 
+/// One coefficient and its position in the `WeightToFeePolynomial`.
+///
+/// One term of polynomial is calculated as:
+///
+/// ```ignore
+/// coeff_integer * x^(degree) + coeff_frac * x^(degree)
+/// ```
+///
+/// The `negative` value encodes whether the term is added or substracted from the
+/// overall polynomial result.
+#[derive(Clone, Encode, Decode)]
+pub struct WeightToFeeCoefficient<Balance> {
+	/// The integral part of the coefficient.
+	pub coeff_integer: Balance,
+	/// The fractional part of the coefficient.
+	pub coeff_frac: Perbill,
```

Comment:
why not just use `Fixed128` instead of three different parameters?

Reply:
In order to have a fully sized integer multiplication. Maybe unnecessary but this structure isn't stored. So size is not an issue.

Reply:
My only complaint is this makes `WeightToFeeCoefficient` harder to use and harder to understand the code.